documentation/c10-classdesign.sil

```diff
@@ -1,20 +1,301 @@
+Let’s do this. Change the line setting the indentation in your custom command…
+
+\begin[type=autodoc:codeblock]{raw}
+     local indent = SILE.settings:get("mypkg.blockindent"):absolute()
+\end{raw}
+
+… and declare the corresponding setting:
+
+\begin[type=autodoc:codeblock]{raw}
+function package:declareSettings ()
+  SILE.settings:declare({
+    parameter = "mypkg.blockindent",
+    type = "measurement",
+    default = SILE.measurement("2em"),
+    help = "Blockquote indentation"
+  })
+end
+\end{raw}
+
+\subsection{Defining raw handlers}
+
+“Raw handlers” allow packages to register new handlers (or callbacks) for use with the \autodoc:environment{raw} environment, which content is read as-is by SILE, without being interpretated.
+This is intended for advanced use cases where you may want to provide a way for users to embed arbitrary content (likely in another syntax), and you will provide the complete parsing and handling for it.\footnote{%
+This may be used to implement a “clever” verbatim environment.
+It is also used, for instance, by the \strong{markdown.sile} 3rd-party collection to embed Markdown or Djot content directly in a (SIL or XML) document.}
+
+You can define your own raw handlers at the Lua level.
+Overloading the \code{registerRawHandlers} package method; and within it, use the \code{self:registerRawHandler} function.
+It takes two parameters: a handler type name, and a function to implement the handler.
+The signature of the handler function is the same as for a SILE command.
+
+Here is a handler that just typesets the content as-is, for you to just get the idea.
+
+\begin[type=autodoc:codeblock]{raw}
+function package:registerRawHandlers ()
+  self:registerRawHandler("mypkg:noop", function(options, content)
+    -- contains everything within the raw environment as unparsed text.
+    local text = content[1]
+    SILE.typesetter:typeset(text)
+  end)
+end
+\end{raw}
+
+\subsection{Loading other packages}
+
+Above, when introducing the \code{_init} method, we left a few placeholder comments.
+Let’s you want to ensure the \autodoc:package{color} package is also loaded, so that the custom \autodoc:command[check=false]{\link} command you implemented can safely invoke it in a \code{SILE.call}.
+
+\begin[type=autodoc:codeblock]{raw}
+function package:_init ()
+  base._init(self)
+  -- Load some dependencies
+  self:loadPackage("color")
+end
+\end{raw}
+
+The \code{self:loadPackage} methods takes as argument a package name, and optionally packages options (as a table).
+
+\subsection{Registering class hooks}
+
+Some packages may provide additional functions that need to be automatically called at various points in the output routine of the document class.
+But let’s return to that topic later, when describing how to set up you own custom class.
+For now, we can conclude our primer on packages, as you should already have all the tools design great packages.
```

Comment:
"all the tools design" > "all the tools to design" likely ^^ Why can't I spell things right, doh.